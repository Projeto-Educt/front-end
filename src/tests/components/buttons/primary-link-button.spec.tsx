import PrimaryLinkButton from '@/components/button/primary-link-button';
import { render, screen } from '@testing-library/react';

describe('PrimaryLinkButton', () => {
  it('should render correctly', () => {
    render(<PrimaryLinkButton href="#">Jest</PrimaryLinkButton>);

    const link = screen.getByRole('link', { name: 'Jest' });

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '#');
  });
  it('Should match snapshot', () => {
    const { container } = render(<PrimaryLinkButton href="#">Jest</PrimaryLinkButton>);

    expect(container).toMatchSnapshot();
  });
});
