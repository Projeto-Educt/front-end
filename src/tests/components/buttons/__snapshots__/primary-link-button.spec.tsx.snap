// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryLinkButton Should match snapshot 1`] = `
<div>
  <div
    class="text-white bg-primary-600 rounded-md h-10 text-center w-full font-bold flex items-center justify-center undefined"
  >
    <a
      class="w-full"
      href="#"
      target="_self"
    >
      Jest
    </a>
  </div>
</div>
`;
